src/components/common/DynamicFields.res

```diff
@@ -1,7 +1,10 @@
 open ReactNative
 open Style
 open RequiredFieldsTypes
-
+type customValidationFunc =
+  | Error(string)
+  | OtherValidation
+  | NoError
```

Comment:
It seems if customValidationFunc is passed you need to call it. Can it be done by making customValidationFunc an optional prop? Instead of making it's type a complex variant?

Reply:
The function is optional, but we need to validate only some data, other data should follow the default data, if we pass a function it will validate all data and this will fail. so we decided to use a variant

Reply:
The variant is needed as per @manideepk90 suggestion or else we need to validate externally if we pass the function

Reply:
sure let's connect tomorrow